parsing/attr_helper.mli

```diff
@@ -19,14 +19,28 @@ type error =
   | Multiple_attributes of string
   | No_payload_expected of string
 
-(** The [string list] argument of the following functions is a list of
-    alternative names for the attribute we are looking for. For instance:
+type identifier = { names: string list; context: string list }
+
+val std_namespace: string -> string list
+val create: ?context:string list -> string -> identifier
+
+(** The [identifier] argument of the following functions is a
+    list of alternative names for the attribute we are looking for and a list of
+    possible other related names that could appear in this context. For instance:
 
     {[
-      ["foo"; "ocaml.foo"]
-    ]} *)
-val get_no_payload_attribute : string list -> attributes -> string loc option
-val has_no_payload_attribute : string list -> attributes -> bool
+      { names=["foo"; "ocaml.foo"]; context=[ "fooo"; "ocaml.fooo"] }
+    ]}
+
+    During attribute identification, a warning is emitted if the nearest
+    name in the union of names and context is in the list of possible
+    names and is at a distance inferior to [attribute_max_distance] *)
```

Comment:
I don't understand what `context` is from this explanation. From the way you use it later I would guess that the words in the context are closed words that we know are *not* mispellings of the attributes (the implementation seems to somewhat confirm), but nothing in the "context" name or in the explanation here allows me to be sure it works this way.